### PR TITLE
[SPARK-36018][CORE][SQL] Some Improvement for Spark Core

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR improve some implement for Spark.

### Why are the changes needed?
This PR improve some implement for Spark.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
Jenkins test.

### CODE REVIEW

**Comment by beliefer** [comment]
retest this please

**Comment by srowen** [comment]
Why are these improvements? there's no detail here

**Comment by beliefer** [comment]
> Why are these improvements? there's no detail here

I found some code seems a little issue. This PR just to make it more reasonable.

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
ping @srowen Could you help me to review this PR?

File: sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala [hunk 1]
```diff
@@ -37,7 +37,7 @@ import org.apache.spark.sql.catalyst.rules.Rule
 class ExperimentalMethods private[sql]() {
 
   /**
-   * Allows extra strategies to be injected into the query planner at runtime.  Note this API
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala:40]: This isn't necessary

**Review by srowen — APPROVED** [review]
OK, these are trivial but OK

File: sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala (hunk 1 above)
Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala:40]: I removed a blank.

Comment (srowen) [reply sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala:40]: Right, I'm saying it doesn't help anything. I'd avoid changes that don't do anything.

Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala]: I reverted it.

**Comment by srowen** [comment]
Merged to master

**Comment by beliefer** [comment]
@srowen Thanks a lot!

**Pull request closed without merging**